scl/checkpoint/plugin.conf

```diff
@@ -51,9 +52,10 @@ block parser checkpoint-parser(prefix('.checkpoint.')) {
                            flags(greedy) delimiters(" ") null("-") dialect(escape-none));
             };
             if {
-                parser { date-parser(format("%Y-%m-%d %H:%M:%S") template("$2 $3")); };
+                parser { date-parser(format("%Y-%m-%dT%H:%M:%S%z",
+                                            "%Y-%m-%dT%H:%M:%S") template("$2")); };
```

Comment:
The order matters in two ways:

1) the more specific format should be parsed first, so that the less specific does not "steal" the parsing from a more specific parsing rule.

2) the one that is expected to be parsed more often should be put first.

For those reasons I've reordered to have the standard ISO timestamp first and the most specific one (e.g. the one that includes the timezone) the very first.

btw, this testcase does not really test that the timestamp was fully parsed, so maybe the testcase passes even if the order in the scl is not correct.

Reply:
btw, I've extracted the reordering to a separate patch so the addition of the new format is the only change in the most important patch in the series.